common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -103,7 +103,11 @@ protected void handleMessage(
           checkAuth(client, msg.appId);
           numBlockIds = 0;
           for (int[] ids: msg.reduceIds) {
-            numBlockIds += ids.length;
+            if (msg.batchFetchEnabled) {
```

Comment:
we don't need the `for` loop when batchFetchEnabled=true
```
if (msg.batchFetchEnabled) {
  numBlockIds = msg.mapIds.length
} else {
  for ...
}
```

Reply:
Copy, also simplified the code in NettyBlockRpcService.